projects/common/src/main/java/dan200/computercraft/shared/pocket/core/PocketServerComputer.java

```diff
@@ -112,20 +113,28 @@ public Map<ResourceLocation, IPeripheral> getUpgrades() {
         return upgrade;
     }
 
+    public @Nullable UpgradeData<IPocketUpgrade> getUpgradeData() {
+        if (upgrade == null) return null;
+        return new UpgradeData<>(
+            upgrade, getUpgradeNBTData()
+        );
+    }
+
     /**
      * Set the upgrade for this pocket computer, also updating the item stack.
      * <p>
      * Note this method is not thread safe - it must be called from the server thread.
      *
      * @param upgrade The new upgrade to set it to, may be {@code null}.
      */
-    public void setUpgrade(@Nullable IPocketUpgrade upgrade) {
-        if (this.upgrade == upgrade) return;
+    public void setUpgrade(@Nullable UpgradeData<IPocketUpgrade> upgrade) {
```

Comment:
So, just for context. This method is not exposed as part of API, and because of this, it was replaced.